docs/source/examples/interactive-development.rst

```diff
@@ -0,0 +1,190 @@
+Start a Development Cluster
+===========================
+
+
+SkyPilot makes interactive development easy on Kubernetes or cloud VMs. It helps you:
+
+#. :ref:`Launch <dev-launch>`: Quickly get a cluster with GPU or other resource requirement with a single command.
+#. :ref:`Autostop <dev-autostop>`: Automatically stop the cluster after some idle time for cost savings.
+#. :ref:`Connect <dev-connect>`: Easily connect to the cluster using the cluster name:
+
+   - :ref:`SSH <dev-ssh>`
+   - :ref:`VSCode <dev-vscode>`
+   - :ref:`Jupyter Notebooks <dev-notebooks>`
+
+.. _dev-launch:
+
+Launch
+------
+
+To launch a cluster with a cheap GPU for development:
+
+.. code-block:: bash
+
+  # Launch a cluster with 1 NVIDIA GPU and sync the local working directory to the
+  # cluster.
+  sky launch -c dev --gpus T4 --workdir .
+
+This can be launched as a pod in your Kubernetes cluster or a cloud VM.
+
+
+
+
+.. note::
+
+  View the supported GPUs with the :code:`sky show-gpus` command.
+
+.. _dev-autostop:
+
+Autostop
+--------
+
+SkyPilot allows you to automatically stop the cluster after a period of idle time to save costs. You can set the autostop time with a single command:
+
+.. code-block:: bash
+
+  # Auto stop the cluster after 5 hours
+  sky autostop -i 300 dev
+
+Or add an additional flag :code:`-i` during the launch:
+
+.. code-block:: bash
+
+  # Launch a cluster with auto stop after 5 hours
+  sky launch -c dev --gpus T4 --workdir . -i 300
+
+For more details of auto stopping, check out: :ref:`auto-stop`. This feature is designed
+to prevent idle clusters from incurring unnecessary costs, ensuring your cluster
+stops automatically, whether it's overnight or throughout the weekend.
+
+
+.. _dev-connect:
+
+Connect
+-------
+
+A user can easily connect to the cluster using your method of choice:
+
+.. _dev-ssh:
+
+SSH
+~~~
+
+SkyPilot will automatically configure the SSH setting for a cluster, so that users can connect to the cluster with the cluster name:
+
+.. code-block:: bash
+  
+  ssh dev
+
+
+.. _dev-vscode:
+
+VSCode
+~~~~~~
+
+A common use case for interactive development is to connect a local IDE to a remote cluster and directly edit code that lives on the cluster. 
+This is supported by simply connecting VSCode to the cluster with the cluster name:
+
+#. Click on the top bar, type: :code:`> remote-ssh`, and select :code:`Remote-SSH: Connect Current Window to Host...`
+#. Select the cluster name (e.g., ``dev``) from the list of hosts.
+#. Open folder: :code:`sky_workdir` and find your code.
```

Comment:
Minor suggestion, feel free to ignore - this one-liner `code --remote ssh-remote+<cluster_name> '/home/sky/sky_workdir'` may be easier to use than clicking in menus

Reply:
Good point! It seems that we can not easily determine the home path for '/home/sky/sky_workdir' on different clouds. I am going to leave this out for now. Will try to add this later.